activation/post_service.go

```diff
@@ -0,0 +1,148 @@
+package activation
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"github.com/spacemeshos/post/config"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/events"
+)
+
+// DefaultPostServiceConfig returns the default config for post service. These are intended for testing.
+func DefaultPostServiceConfig() PostServiceConfig {
+	cfg := PostServiceConfig{
+		PostServiceCmd:  "/bin/service",
+		DataDir:         config.DefaultDataDir,
+		NodeAddress:     "http://127.0.0.1:9093",
+		PowDifficulty:   config.DefaultConfig().PowDifficulty,
+		PostServiceMode: "light",
+	}
+
+	return cfg
+}
+
+// MainnetPostServiceConfig returns the default config for mainnet.
+func MainnetPostServiceConfig() PostServiceConfig {
+	cfg := PostServiceConfig{
+		PostServiceCmd:  "/bin/service",
+		DataDir:         config.DefaultDataDir,
+		NodeAddress:     "http://127.0.0.1:9093",
+		PowDifficulty:   config.MainnetConfig().PowDifficulty,
+		PostServiceMode: "fast",
+	}
+
+	return cfg
+}
+
+type PostServiceConfig struct {
+	PostServiceCmd string `mapstructure:"post-opts-post-service"`
+
+	DataDir         string        `mapstructure:"post-opts-datadir"`
+	NodeAddress     string        `mapstructure:"post-opts-node-address"`
+	PowDifficulty   PowDifficulty `mapstructure:"post-opts-pow-difficulty"`
+	PostServiceMode string        `mapstructure:"post-opts-post-service-mode"`
+}
+
+// PostService manages a local post service.
+type PostService struct {
+	logger *zap.Logger
+
+	pidMtx sync.RWMutex
+	pid    int // pid of the running post service.
```

Comment:
It's just a single value that needs to be set atomically. How about:
```suggestion
	pid atomic.Int64 // pid of the running post service.
```

Reply:
Yeah, it is also only used in a single test and shouldn't be used anywhere else. I'll get rid of the `Pid()` method as well.